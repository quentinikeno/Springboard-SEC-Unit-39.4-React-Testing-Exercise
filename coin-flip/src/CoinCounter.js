import React from "react";
import Coin from "./Coin";

const CoinCounter = ({ sides }) => {
	const [side, setSide] = null;
	const [totalFlips, setTotalFlips] = 0;
	const [totalTails, setTotalTails] = 0;
	const handleClick = () => {
		if (Math.random() > 0.5) {
			setSide("tails");
			setTotalTails(totalTails + 1);
		} else {
			setSide("heads");
		}
		setTotalFlips(totalFlips + 1);
	};

	return (
		<div>
			<h1>Let's Flip a Coin!</h1>
			<Coin imgSrc={sides[side]} />
			<button onClick={handleClick}>Flip Me!</button>
			<p>
				Out of {totalFlips} flips, there have been{" "}
				{totalFlips - totalTails} heads and {totalTails} tails.
			</p>
		</div>
	);
};

CoinContainer.defaultProps = {
	sides: {
		heads: {
			imgSrc: "https://tinyurl.com/react-coin-heads-jpg",
		},

		tails: {
			imgSrc: "https://tinyurl.com/react-coin-tails-jpg",
		},
	},
};

export default CoinCounter;
