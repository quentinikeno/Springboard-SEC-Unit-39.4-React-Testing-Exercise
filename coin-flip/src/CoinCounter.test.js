import React from "react";
import { render } from "@testing-library/react";
import CoinCounter from "./CoinCounter";

// smoke test
it("renders without crashing", function() {
	render(<CoinCounter />);
});

// snapshot test
it("matches snapshot", function() {
	const { asFragment } = render(<CoinCounter />);
	expect(asFragment()).toMatchSnapshot();
});
