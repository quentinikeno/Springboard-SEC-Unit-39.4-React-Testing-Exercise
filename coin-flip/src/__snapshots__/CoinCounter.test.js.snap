// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Let's Flip a Coin!
    </h1>
    <button>
      Flip Me!
    </button>
    <p>
      Out of 0 flips, there have been 0 heads and 0 tails.
    </p>
  </div>
</DocumentFragment>
`;
